src/components/Image/Image.tsx

```diff
@@ -7,34 +14,104 @@ export interface SourceSet {
 
 export type CrossOrigin = 'anonymous' | 'use-credentials' | '' | undefined;
 
+export interface State {
+  hasError: boolean;
+  hasLoaded: boolean;
```

Comment:
has Loaded is only set to true if the image loads properly. This is used to change the style on the image.